app/views/_controls.html.erb

```diff
@@ -9,10 +9,10 @@
   <div class="row">
     <div class="col-md-4"></div>
     <div class="col-md-1">
```

Comment:
Try `col-xs-push-1 col-sm-7 col-sm-push-3 col-md-5 col-md-push-4` (to match the search box) and move both links into the save div.
This should fix up the extra scrolling and keep the links inline.